### PR TITLE
[Backport 2023.02.xx] #9369 Setting up readthedocs build.os

### LINKED ISSUE #9369: Fix readthedocs build.os
Accrordingly to the readthedocs blog 

https://blog.readthedocs.com/use-build-os-config/

We need to migrate our configuration to make work against readthedocs build as soon as possible. Final deadline is October 16th, but anyway we will have several brownout at Sep 18th and Oct 2nd that may cause troubles.

We did a first migration here to v2:

https://github.com/geosolutions-it/MapStore2/issues/9333
https://github.com/geosolutions-it/MapStore2/pull/9297

But we had an issue, discussed [here](https://github.com/readthedocs/readthedocs.org/issues/10585) so we had to to comment `build.os` parts. Now `build.os` is going to become mandatory, so we have to find a solution.

#### Brownout Schedule:

```
Monday, August 28, 2023: Do the first brownout (temporarily enforce this deprecation) for 12 hours: 00:01 PST to 11:59 PST (noon)

Monday, September 18, 2023: Do a second brownout (temporarily enforce this deprecation) for 24 hours: 00:01 PST to 23:59 PST (midnight)

Monday, October 2, 2023: Do a third and final brownout (temporarily enforce this deprecation) for 48 hours: 00:01 PST to October 3, 2023 23:59 PST (midnight)

Monday, October 16, 2023: Fully remove support for building documentation using “build.image” on the configuration file
```

### PR DESCRIPTION
**Pull request by offtherailz**
[Backport 2023.02.xx] #9369 Setting up readthedocs build.os

### CODE REVIEW

**Comment by offtherailz** [comment]
Waiting for backport approval

**Review by tdipisa — APPROVED** [review]

**Pull request merged**
